components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -345,6 +389,58 @@ void AIChatTabHelper::CheckPDFA11yTree() {
   OnPDFA11yInfoLoaded();
 }
 
+void AIChatTabHelper::FetchSearchQuerySummary(
+    FetchSearchQuerySummaryCallback callback) {
+  if (!IsBraveSearchSERP(GetPageURL())) {
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
+
+  auto internal_callback =
+      base::BindOnce(&AIChatTabHelper::OnSearchSummarizerKeyFetched,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+  GetSearchSummarizerKey(web_contents(), std::move(internal_callback));
+}
+
+void AIChatTabHelper::OnSearchSummarizerKeyFetched(
```

Comment:
Should this part be in shared code (via `ConversationDriver`) so that iOS can use it? The platform-specific code seems to be searching the DOM for the meta tag, which we could keep here.

Reply:
@petemill I agreed but could we do a quick follow-up after this PR?

Reply:
ok if you have a reason to move it afterwards

Reply:
https://github.com/brave/brave-browser/issues/40741